contracts/cw20-merkle-airdrop/src/helpers.rs

```diff
@@ -0,0 +1,60 @@
+use crate::msg::{SignatureInfo, SignedClaimMsg};
+use crate::ContractError;
+use bech32::ToBase32;
+use cosmwasm_std::{from_binary, to_vec, Binary, Deps, DepsMut, MessageInfo};
+use ripemd::{Digest as RipDigest, Ripemd160};
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+use sha2::{Digest as ShaDigest, Sha256};
+use std::convert::TryInto;
+
+pub fn verify_cosmos(
+    deps: Deps,
+    claim_msg: &SignedClaimMsg,
+    sig: &CosmosSignature,
+) -> Result<bool, ContractError> {
+    let msg_raw = to_vec(claim_msg)?;
+    let hash = Sha256::digest(&msg_raw);
+
+    deps.api
+        .secp256k1_verify(
+            hash.as_ref(),
+            sig.signature.as_slice(),
+            sig.pub_key.as_slice(),
+        )
+        .map_err(|_| ContractError::VerificationFailed {})
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct CosmosSignature {
+    pub pub_key: Binary,
+    pub signature: Binary,
+}
+
+pub fn verify_external_address(
+    deps: &DepsMut,
+    info: &MessageInfo,
+    hrp: String,
+    sig: &SignatureInfo,
+) -> Result<String, ContractError> {
+    // check if signature is correct
+    let signature: CosmosSignature = from_binary(&sig.signature).unwrap();
```

Comment:
You have variables `sig` and `signature` which are completely separate structures.
```suggestion
    let cosmos_signature: CosmosSignature = from_binary(&sig.signature).unwrap();
``` 
?